### PR TITLE
Fix/testimonials issues

### LINKED ISSUE #58: Testimonials Issues
1. All the logic for testimonials should be moved to a portable `useTestimonials` hook.
2. The way styled-components are declared here is overkill. Prefer the `sx` prop.

_Originally posted by @pkrawc in https://github.com/rentalutions/marketing/pull/44#discussion_r578546028_

___

Probably don't need to check for getComputedStyle in a useEffect?

_Originally posted by @pkrawc in https://github.com/rentalutions/marketing/pull/44#discussion_r578547171_

___

We can probably use framer-motion's useTransform for this.

```javascript
const motionLevel = useSpring(level)
const opacity = useTransform(motionLevel, [0,3], [0.5, 1])
const scale = useTransform(motionLevel, [0, 3], [0.5, 0.75])
useEffect(() => {
  motionLevel.set(level)
}, [level])
```

_Originally posted by @pkrawc in https://github.com/rentalutions/marketing/pull/44#discussion_r578574776_

### PR DESCRIPTION
**Pull request by henriquecfreitas**
Resolves #58

### CODE REVIEW

File: src/components/elements/LevelEffect/index.js [hunk 1]
```diff
@@ -0,0 +1 @@
+export { default as useLevelEffect } from "./use-level-effect"
```
Comment (quaardvark) [inline src/components/elements/LevelEffect/index.js:1]: Why not put it in the animation utils file we already have? Let's please do that.

File: src/components/molecules/Cards/index.js [hunk 2]
```diff
@@ -0,0 +1,3 @@
+export { default as useCards } from "./use-cards"
+export { default as useCardsControl } from "./use-cards-control"
+export { default as CardsControl } from "./cards-control"
```
Comment (quaardvark) [inline src/components/molecules/Cards/index.js:3]: Please, let's stop using re-exports. They are pointless and make it harder to trace / jump between implementations. Just import directly from file with implementation. This packages are internal anyway.

Furthermore, I don't understand why we need this as a molecule. Can you envision any other context it's going to be used in this way? I agree with the card visual as a molecule/element, but Cards with the whole logic package? 

Let's please contain the files to the Testimonials organism for now

File: src/components/molecules/Carousel/use-carousel.js [hunk 3]
```diff
@@ -1,11 +1,11 @@
-import React, { useCallback, useEffect, useState } from "react"
+import { useCallback, useEffect, useState } from "react"
```
Comment (quaardvark) [inline src/components/molecules/Carousel/use-carousel.js:1]: Same argument applies here, let's contain the hook to the organism

File: src/components/organisms/Testimonials/carousel/testimonials-carousel-item.js [hunk 4]
```diff
@@ -11,12 +12,12 @@ function TestimonialsCarouselItem({
...
-    if (level === 2) return [0.75, 0.625]
-    if (level === 1) return [0.5, 0.75]
-    return [0, 1]
-  })()
+  const { opacity, scale } = useLevelEffect({
+    level,
+    levelRange: [0, 2],
+    opacityRange: [0, 0.75],
+    scaleRange: [1, 0.625],
+  })
```
Comment (quaardvark) [inline src/components/organisms/Testimonials/carousel/testimonials-carousel-item.js:20]: Can you explain what was achieved here? I don't see the animations on the preview

**Review by quaardvark — CHANGES_REQUESTED** [review]
@pkrawc can you elaborate what did you mean by splitting the logic into hooks? Do you envision having them as reusable / standalone components? Or was it just for readability sake?

I any case, I think that we should refrain from declaring them as separate components, - let's just put the files alongside the Testimonials implementation for now. If they will eventually be reusable / standalone components - then we will move them. They are not right now and are just making the components list harder to assess.

File: src/components/elements/LevelEffect/index.js (hunk 1 above)
Comment (henriquecfreitas) [reply src/components/elements/LevelEffect/index.js:1]: Sure!
I also thought that it might be worthy to create an `effects` package on `@rent_avail`

**Comment by pkrawc** [comment]
> @pkrawc can you elaborate on what did you mean by splitting the logic into hooks? Do you envision having them as reusable/standalone components? Or was it just for readability's sake?

> I any case, I think that we should refrain from declaring them as separate components, - let's just put the files alongside the Testimonials implementation for now. If they will eventually be reusable/standalone components - then we will move them. They are not right now and are just making the components list harder to assess.

Mostly for readability, a little bit for portability. It makes sense to me to house the styling and layout of the component in one file and the rendering logic in another (hook). Eg. the code standards in the readme layout this pattern.

@quaardvark

**Comment by quaardvark** [comment]
@henriquecfreitas do you know why the build is failing here?

File: src/components/organisms/Testimonials/carousel/testimonials-carousel-item.js (hunk 4 above)
Comment (quaardvark) [reply src/components/organisms/Testimonials/carousel/testimonials-carousel-item.js:20]: @henriquecfreitas Am I testing something wrong? I can't see animation on - https://marketing-git-fix-testimonials-issues-rent-avail.vercel.app/info/listings-dev-test

**Comment by henriquecfreitas** [comment]
> @henriquecfreitas do you know why the build is failing here?

Wrong export.
Already fixed, thanks!

File: src/components/organisms/Testimonials/carousel/testimonials-carousel-item.js (hunk 4 above)
Comment (quaardvark) [reply src/components/organisms/Testimonials/carousel/testimonials-carousel-item.js:20]: @henriquecfreitas

Comment (quaardvark) [reply src/components/organisms/Testimonials/carousel/testimonials-carousel-item.js:20]: @henriquecfreitas are we planning to address this question?

**Comment by pkrawc** [comment]
Pretty sure this is stale and doesn't actually need merging, but is there any problem with me doing so? There are no conflicts with staging that I can see.

@henriquecfreitas @quaardvark

**Comment by henriquecfreitas** [comment]
> Pretty sure this is stale and doesn't actually need merging, but is there any problem with me doing so? There are no conflicts with staging that I can see.
> 
> @henriquecfreitas @quaardvark

Most of the work from this PR is isolating code from `TestimonialsCards` to `useTestimonialsCards`. I wouldn't merge it yet, knowing that this work will be lost as soon as `Slider` gets introduced (bc the logic from `useTestimonialsCards` will be moved to `useSlider` and the slider layout from `TestimonialsCards` will be moved to `Slider`).

I'm already working on the `Slider` component, in order to use it for the Article List component (https://www.figma.com/file/tyf4c9SVsRKSqxTEOyTDkb/?node-id=216%3A0).

The issues related to `TestimonialsCards` will be fixed on that branch. Then we can rebase and rethink this PR in order to address the issues related to `TestimonialsCarousel`. I'll mark this as a WIP and resume working on it as soon as possible.

**Comment by henriquecfreitas** [comment]
@pkrawc @quaardvark I removed the old commits from this tree since PR #94 already addressed those card/slider related issues (and bc they were very outdated). I left in this PR the changes related to the carousel version (lint fixes and minor adjustments)
Now there isn't much changes left here, but I vote to merge them and close the issue #58

**Comment by henriquecfreitas** [comment]
I'm closing this PR because all the changes it proposed are being applied by PRs #94 and  #105

**Pull request closed without merging**